core/src/main/java/com/aem/demo/core/models/NewsCardModel.java

```diff
@@ -39,7 +44,62 @@ public class NewsCardModel {
     @Default(values = DEFAULT_IMAGE)
     private String image;
 
+//    private int likes;
+
+    @Inject
+    private SlingHttpServletRequest servletRequest;
+
+    @SlingObject
+    private ResourceResolver resourceResolver;
+
+
     public String getPath() {
-        return resource.getPath().replace("/jcr:content","");
+        return resource.getPath().replace("/jcr:content", "");
+    }
+
+    public String getFullPath() {
+        return resource.getPath();
+    }
+
+
+    public int  getLikes() {
+        int likes = 0;
+
+        String rootPath = resource.getPath();
```

Comment:
Man, are u serious?))
You are retrieving a path from the resource only to get all the same resource with all the same path?))